PROJECT: TIC TAC TOE

macnick opened the pull request:
We have made a simple Tic Tac Toe game you that you can play in your browser!

The users can
1. . Fill in their names in the boxes
2. Click on Play to start a new game
3. If there is a winner or the game finishes with a tie click on the popup message to make it go away
4. Click on Play to play again

We created separate modules for Board and Player. Also the whole system is an MVC design. The logic is in the controller module.  The view deals only with  screen drawing. 

Please review our code

macnick committed:
Initial linter setup

macnick committed:
Add some javascript

macnick committed:
Fix 3 modules for mode;, view, controller

macnick committed:
Create html board

macnick committed:
Fix displayboard to show the clicked boxes

macnick committed:
Create Player module

addod19 committed:
added reset method to view

macnick committed:
Allow players to add their mark to the DOM

addod19 committed:
added check winner method

addod19 committed:
added reset method to view

macnick committed:
Highlight the name of the current player

addod19 committed:
added reset button and reset function

addod19 committed:
added reset function and button

addod19 committed:
check winner function added

addod19 committed:
resolving merge conflicts

macnick committed:
Create method to heck for winner

macnick committed:
Merge branch 'modules' of https://github.com/macnick/Tic-Tac-Toe-JS into modules

macnick committed:
Move reset button to the middle and cleanup CSS

macnick committed:
Fix reset button and game over

macnick committed:
Fix reset button and game over

macnick committed:
Merge with origin

macnick committed:
Add check for tie and end of game

macnick committed:
Add fields to input player names

macnick committed:
Add fields to input player names

macnick committed:
Add input fields for player names

macnick committed:
Add input fields for player names

macnick committed:
Integrate input fields with player name labels

macnick committed:
Merge branch 'modules' of https://github.com/macnick/Tic-Tac-Toe-JS into modules

macnick committed:
clean up commented-out code

addod19 committed:
added show modal to view

addod19 committed:
added close modal function to view

addod19 committed:
modified indexjs

addod19 committed:
add model window

macnick committed:
Add higlight to winning combination

macnick committed:
Add winning modal

macnick committed:
Fix eslint errors

addod19 committed:
update

addod19 committed:
fix linter errors

macnick committed:
Change modal top margin

macnick committed:
Add README file

addod19 committed:
added images folder with screen shots

addod19 committed:
Merge branch 'modules' of https://github.com/macnick/Tic-Tac-Toe-JS into modules

addod19 committed:
update readme

addod19 committed:
Update README.md

bellom commented on assets/index.js:
```diff
@@ -0,0 +1,164 @@
+/* eslint-disable import/extensions */
+import Player from './player.js';
+import Table from './board.js';
+
```
clear code!
kindly move the DOM functions to another file.
![image](https://user-images.githubusercontent.com/31897434/76318591-39c9a700-62de-11ea-8586-973cc1f2aad9.png)

macnick replied:
Done!

bellom commented on assets/index.js:
```diff
@@ -0,0 +1,164 @@
+    }
+  };
+
+  const isItATie = () => t.t.every((pos) => pos != null);
```
[Optional]
Am not good in naming too 😄 but you can name this `checkTie` right?
seems your function name is a complete sentence 🙂

macnick replied:
Done!

bellom reviewed (changes requested):
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**_1. Kindly check out my comments below;_**

**_2. Kindly make typing of user's name required_**

**_3. Kindly add `reset` button;_**
![image](https://user-images.githubusercontent.com/31897434/76319606-c2951280-62df-11ea-9b24-474d842edf58.png)


**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

macnick commented:
Thank you @bellom for your kind and prompt review. 👍 
We fixed the 2 things you suggested. 
1. Moved the DOM module to a separate file.
2. Renamed the isItATie to checkTie which makes more sense. 

Regarding the requirements 2 and 3 we have some objection. Please read and think about them.  

According to instructions :
- "Clean up the interface to **allow** players to put in their names,"
We have to **allow** the player to add a name, not to **enforce** the name input. As it is, the game allows the player to do whatever he likes. More freedom. 
- " include a button to start/restart the game" We already have a button to start/restart the game. We do not need a second button.
- "and add a display element that congratulates the winning player!"
After a player wins he/she gets a message "Winner is PlayerName". Do we really have to change it to read: "Congratulations PlayerName you won!" ? 

Thank you for your patience. 

Please review our code and play the game with a friend. 🙏

macnick committed:
Move view to another file

bellom commented:
Hey @macnick, 

Thanks for the professional reply.
2 and 3 should have been tag as optional 🙏 

Let's move to next project guys 👍 

Have you guys submitted another `CRR`?

addod19 commented:
> Hey @macnick,
> 
> Thanks for the professional reply.
> 2 and 3 should have been tag as optional 🙏
> 
> Let's move to next project guys 👍
> 
> Have you guys submitted another `CRR`?

Yes we have.

Oghenebrume50 reviewed (approved):
### STATUS: APPROVED 

Hi, Team this is awesome, all corrections from the previous review were made project looks good now, please merge and move on :+1: **This is approved** :slightly_smiling_face: 

**RAPHAEL NORIODE** :sunglasses:

addod19 commented:
> ### STATUS: APPROVED
> Hi, Team this is awesome, all corrections from the previous review were made project looks good now, please merge and move on 👍 **This is approved** 🙂
> 
> **RAPHAEL NORIODE** 😎

Thank you for the quick review 👐
